ckanext/canada/tabulate.py

```diff
@@ -0,0 +1,311 @@
+
+    def __mangle__prepare_dialect(self, stream):
+        # Get sample
+        # Copied from tabulator.pasrers.csv
+        # Needed because we cannot call parent private method while mangling.
+        sample = []
+        while True:
+            try:
+                sample.append(next(stream))
+            except StopIteration:
+                break
+            if len(sample) >= CSV_SAMPLE_LINES:
+                break
+
+        # Get dialect
+        # Copied from tabulator.pasrers.csv
+        # Needed to get the guessed dialect. Using mangling for protected properties.
+        try:
+            separator = b'' if six.PY2 else ''
+            delimiter = self._CSVParser__options.get('delimiter', ',\t;|')
+            dialect = csv.Sniffer().sniff(separator.join(sample), delimiter)
+            if not dialect.escapechar:
+                dialect.doublequote = True
+        except csv.Error:
+            class dialect(csv.excel):
+                pass
+        for key, value in self._CSVParser__options.items():
+            setattr(dialect, key, value)
+        # https://github.com/frictionlessdata/FrictionlessDarwinCore/issues/1
+        if getattr(dialect, 'quotechar', None) == '':
+            setattr(dialect, 'quoting', csv.QUOTE_NONE)
+
+        self._CSVParser__dialect = dialect
+
+        if self.static_encoding:
+            log.debug('Using static encoding for csv: %s', self.static_encoding)
+        log.debug('Using Static Dialect for csv: %r', self.static_dialects['csv'])
+
+        # Xloader does not use Goodtables, and enters the Stream Parser fairly directly.
+        # So we need to check and raise here. self.static_dialects will always exist in this mangle.
+        _check_dialect_and_raise(self._CSVParser__dialect, CanadaCSVDialect(self.static_dialects['csv']), compare_obj=True)
+
+        return sample, CanadaCSVDialect(self.static_dialects['csv'])
+
+
+def _get_dialects():
+    dialects = config.get('ckanext.canada.tabulator_dialects', None)
+    if dialects:
+        return json.loads(dialects)
+    return None
+
+
+def _get_encoding():
+    return config.get('ckanext.canada.tabulator_encoding', None)
+
+
+def _check_dialect_and_raise(guessed_dialect, static_dialect, compare_obj=False):
+    guessed_delimiter_var = 'delimiter'
+    guessed_quoteChar_var = 'quoteChar'
+    guessed_doubleQuote_var = 'doubleQuote'
+
+    if compare_obj:
+        # a bit nasty, but need to compare from __mangle__prepare_dialect
+        guessed_dialect = guessed_dialect.__dict__
+        static_dialect = static_dialect.__dict__
+        guessed_delimiter_var = 'delimiter'
+        guessed_quoteChar_var = 'quotechar'
+        guessed_doubleQuote_var = 'doublequote'
+
+    if guessed_dialect and static_dialect:
+        if guessed_delimiter_var in guessed_dialect and guessed_dialect[guessed_delimiter_var] != static_dialect['delimiter']:
+            raise TabulatorException('{0}{1}{2}'.format(
```

Comment:
This is a little odd, you could do something like
```suggestion
            raise TabulatorException(''.join(
```
instead, or even better: `ERROR_SPLITTER.join(` with only two parameters.